### PR TITLE
Fix regressions in link UI

### PR DESCRIPTION
**Pull request by jasmussen**
Recently we merged in some improvements to the design of input fields. As part of that, some regressions to the link UI on buttons and link modals were introduced. This PR fixes those.

Screenshots:

screen shot 2018-04-17 at 09 51 45

screen shot 2018-04-17 at 10 27 49

screen shot 2018-04-17 at 10 27 56

screen shot 2018-04-17 at 10 28 03

As part of this, I did some major cleanup too, to use more SCSS variables stored in the variables stylesheet. I also added some admin scheme color to the selected item in the suggest dropdown. In fact, quite a bit of CSS was messy, duplicate and redundant, so this is quite a nice cleanup to get in. But it also means that it's a fair bit of deep surgery, and would be good to test. 

@karmatosed I know you've filed a few tickets around link interface improvements. Does this PR address some of those?

### CODE REVIEW

**Comment by karmatosed** [comment]
This is great, thanks for picking up this. The one we had open about https://github.com/WordPress/gutenberg/issues/6107 : unifying on buttons, is still looking like it has the old interface. That is the one I would love to see changed :)

**Review by karmatosed — APPROVED** [review]

**Comment by jasmussen** [comment]
Thanks Tammie, responded to that other ticket.

File: blocks/url-input/style.scss [hunk 1]
```diff
@@ -30,35 +27,43 @@
 	}
 }
 
+// Suggestions
 .blocks-url-input__suggestions {
 	position: absolute;
-	top: 100%;
-	left: 0;
-	right: -64px; // to match the link modal borders
+	z-index: 0;	// place underneath the input field so focus isn't cropped
```
Comment (youknowriad) [inline blocks/url-input/style.scss]: Should we use the z-index helper instead?

**Comment by jasmussen** [comment]
🔥

**Pull request merged**